packages/ui/src/components/ButtonV2/index.tsx

```diff
@@ -0,0 +1,293 @@
+import styled from '@emotion/styled'
+import type {
+  ButtonHTMLAttributes,
+  ComponentProps,
+  ForwardedRef,
+  ReactNode,
+} from 'react'
+import { forwardRef } from 'react'
+import type { SENTIMENTS } from '../../theme'
+import { Icon } from '../Icon'
+import { Loader } from '../Loader'
+
+type SENTIMENT = (typeof SENTIMENTS)[number]
+
+// SIZE
+export const SIZE_HEIGHT = {
+  large: 48,
+  medium: 40,
+  small: 32,
+} as const
+type ButtonSize = keyof typeof SIZE_HEIGHT
+export const SIZE_SPACING_KEY = {
+  large: 2,
+  medium: 1.5,
+  small: 1,
+} as const
+export const buttonSizes = Object.keys(SIZE_HEIGHT) as ButtonSize[]
+
+// FOCUS RING
+const FOCUS_RING_KEY = {
+  danger: 'focusDanger',
+  info: 'focusInfo',
+  neutral: 'focusNeutral',
+  primary: 'focusPrimary',
+  success: 'focusSuccess',
+  warning: 'focusWarning',
+} as const
+
+// VARIANTS
+type StyledButtonProps = Required<
+  Pick<
+    ButtonProps,
+    'size' | 'sentiment' | 'disabled' | 'iconPosition' | 'fullWidth'
+  >
+>
+const StyledButton = styled('button', {
+  shouldForwardProp: prop =>
+    !['size', 'sentiment', 'iconPosition', 'fullWidth'].includes(prop),
+})<StyledButtonProps>`
+  display: inline-flex;
+  height: ${({ size }) => `${SIZE_HEIGHT[size]}px`};
+  padding: 0 ${({ theme, size }) => theme.space[SIZE_SPACING_KEY[size]]};
+  flex-direction: ${({ iconPosition }) =>
+    iconPosition === 'right' ? 'row-reverse' : 'row'};
+  gap: ${({ theme }) => theme.space['1']};
+  border-radius: ${({ theme }) => theme.radii.default};
+  align-items: center;
+  cursor: ${({ disabled }) => (disabled ? 'not-allowed' : 'pointer')};
+  width: ${({ fullWidth }) => (fullWidth ? '100%' : 'auto')};
+  justify-content: center;
+  outline-offset: 2px;
```

Comment:
I think we can use space here from theme

```suggestion
  outline-offset: ${({ theme }) => theme.space['0.25']};
```